Implemented Sequence Detection with Interface

MANASA-NUKALA opened the pull request:
- Implemented SequenceSolver to detect arithmetic and geometric sequences.
- Created SequenceDetector interface for modularity.
- Added user input handling to allow dynamic sequence detection.

NUKALA-MANASA committed:
Added SequenceSolver with user input handling

yuripourre commented:
Hi @MANASA-NUKALA, first of all thank you for the contribution.

I like your idea of having an interface.

To make it more reusable I added some comments. Please let me know what you think.

MANASA-NUKALA commented:
Hi @yuripourre , I appreciate your feedback! However, I can't see any comments on the PR.
Could you please resubmit them or let me know what changes you suggest? Thanks!

yuripourre commented:
> Hi @yuripourre , I appreciate your feedback! However, I can't see any comments on the PR. Could you please resubmit them or let me know what changes you suggest? Thanks!

Oops, my bad, I forgot to submit the review

MANASA-NUKALA committed:
Moved main logic to a separate file and implemented enum-based result

MANASA-NUKALA commented:
Hi @yuripourre ,  
I have moved the main logic to a separate file and also implemented the enum-based result as suggested.  
However, I couldn't get the unit tests to work since I'm new to Maven and JUnit. I tried my best but couldn't figure it out. Let me know if any further changes are required. Thanks!

yuripourre reviewed (commented):
The PR is much better now, I have one last ask.

Is it possible to move the example file to outside the main package?

Maybe moving it to /src/test/examples

Other than that looks good.

yuripourre reviewed (changes requested):
The PR is much better now, I have one last ask.

Is it possible to move the example file to outside the main package?

Maybe moving it to /src/test/examples

Other than that looks good.

MANASA-NUKALA committed:
Moved example file to correct location

MANASA-NUKALA commented:
I initially misunderstood the request and moved the Main.java file to /src/main/java/examples/. But I have now corrected it by moving them to /src/test/examples as per the actual requirement.

yuripourre commented:
@MANASA-NUKALA thanks for the update, I will check it properly later this week.

yuripourre commented on src/main/java/com/harium/suneidesis/math/Main.java:
```diff
@@ -0,0 +1,19 @@
+package com.harium.suneidesis.math;
```
Can you move this class to the same folder as the other examples?

yuripourre commented:
Hi @MANASA-NUKALA, my apologies, I got confuse on my last message. I thought the example classes were in the `/src/test/examples` folder, but they are actually at the `/src/main/java/examples`.

You can leave the previous examples in the original folder: `/src/main/java/examples`.

But can you move your example there? [src/main/java/com/harium/suneidesis/math/Main.java](https://github.com/Harium/suneidesis/pull/151/files/31d678d109a632372c9599fc02c0eb767bbc788e#diff-bcc363455a947f1b0e91acc85c5b80b3901ef63feb1585a7e0895f276a6736b8). Then this PR is good to go.



No class with main should be at `src/main/java/com` because those classes are packaged and added to the .jar file so it adds overhead to whoever uses this library.

MANASA-NUKALA committed:
Moved example file to the correct folder as requested

MANASA-NUKALA commented:
Hi @yuripourre, I first moved Main.java to the examples folder and then relocated the examples folder to /src/main/java/examples as requested. Let me know if any further changes are needed. Thanks!

yuripourre reviewed (changes requested):
Do you mind adding unit tests for the SequenceDetector?

And finally, can you rename the file Main to SequenceDetectorExample?

MANASA-NUKALA commented:
HI @yuripourre ,Thank you for your feedback! I understand the importance of adding unit tests, but since I'm new to Maven and JUnit, I tried implementing them but couldn't get them to work correctly.
